pkg/build/golang/deps.go

```diff
@@ -48,6 +49,14 @@ func parseDependenciesFromDocker(cli *client.Client, imageID string) (map[string
 		return nil, err
 	}
 
+	defer func() {
+		// Remove container
+		err = cli.ContainerRemove(context.Background(), res.ID, types.ContainerRemoveOptions{Force: true})
+		if err != nil {
+			fmt.Printf("error while removing container %s: %v", res.ID, err)
```

Comment:
@raulk @nonsense when deferring, what's the best practice to check for errors? I decided to just print out a message about it. I could also replace the return value in case there is an error here too. Wdyt?

Reply:
I think in this case, it's best for the return value to be replaced.

When you want to `handle` an error (which half of the times actually means just logging it, then better to use a logger, rather than fmt.Printf, because the logger is structured, and has additional information - timestamp, optional context, caller location, etc.)

Reply:
Actually on a second thought about this particular case - this is the `cleanup` phase. If nothing returned an error until the defer statement, then we get the dependencies correctly. Then if the cleanup phase fails for some reason, we can also consider it just as a warning, and move on (although something might fail in the future).

Therefore I think it is just as valid to just log the error, and not replace the returned value - whatever you decide.

Another point is that if something in the function does return an error and then you have another error in the cleanup phase - you would overwrite the first error.

Reply:
@nonsense yes, I completely agree with you about the error overwriting and that's why I didn't replace it on the first turn. I updated it to be a  `log.Warnf`. It doesn't worry me much if the cleanup fails even if something Docker-related might fail in the future, we will get the error afterwards anyways.